tests/boilerplate_client/boilerplate_cmd.py

```diff
@@ -1,11 +1,10 @@
 import struct
-from typing import Tuple
+from typing import Tuple, Generator
+from contextlib import contextmanager
 
-from ragger.backend import BackendInterface
-from ragger.error import ExceptionRAPDU
+from ragger.backend.interface import BackendInterface, RAPDU
 
-from boilerplate_client.boilerplate_cmd_builder import BoilerplateCommandBuilder, InsType
-from boilerplate_client.exception import DeviceException
+from boilerplate_client.boilerplate_cmd_builder import BoilerplateCommandBuilder
```

Comment:
I think that `BoilerplateCommandBuilder` is an unneeded level of abstraction. Adding quite a lot of clumsiness without a lot of perks? Or with perks I don't understand?
Also, I find the chunkify and sign_tx function in `boilerplate_cmd_builder.py` really difficult to understand compare to what it does. Could we change this to a simpler version like what is done in Hedera, Solana, Eos, etc ?

Reply:
Overhauled